Add a condition to include or exclude the processing of nitrate depending on the model's output

zhanglikate opened the pull request:
1. Add a condition to include or exclude the processing of nitrate depending on the model's output
2. Add  XML control files and TXT files for UFS-Aerosols model output.

zhanglikate committed:
Add a condition to include or exclude the processing of nitrate,
depending on the model's output

WenMeng-NOAA commented on parm/postcntrl_gefs_aerosol.xml:
```diff
@@ -10,64 +10,997 @@
     <prod_status>oper</prod_status>
     <data_type>fcst</data_type>
-    <gen_proc_type>fcst</gen_proc_type>
+    <gen_proc_type>ens_fcst</gen_proc_type>
```
@zhanglikate The "ens_fcst" is not an official type of generating process at 
https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-3.shtml
Please revert it.

WenMeng-NOAA commented on parm/postcntrl_gefs_aerosol.xml:
```diff
@@ -10,64 +10,997 @@
     <time_range_unit>hour</time_range_unit>
     <orig_center>nws_ncep</orig_center>
-    <gen_proc>gfs_avn</gen_proc>
+    <gen_proc>gefs</gen_proc>
```
@zhanglikate The "gefs" can't be identified as forecast generating process at
https://www.nco.ncep.noaa.gov/pmb/docs/on388/tablea.html
Please revert it.

zhanglikate replied:
May I know which GEFS control file that I need to following to add the aerosols output for GEFSv13? Thanks.

I am following postcntrl_gefs.xml to add the aerosols output?

zhanglikate commented:
May I know how to revert it? Thanks.

Kate


> On Aug 31, 2023, at 12:59 PM, WenMeng-NOAA ***@***.***> wrote:
> 
> 
> @WenMeng-NOAA commented on this pull request.
> 
> In parm/postcntrl_gefs_aerosol.xml <https://github.com/NOAA-EMC/UPP/pull/768#discussion_r1312080778>:
> 
> > @@ -10,64 +10,997 @@
>      <sigreftime>fcst</sigreftime>
>      <prod_status>oper</prod_status>
>      <data_type>fcst</data_type>
> -    <gen_proc_type>fcst</gen_proc_type>
> +    <gen_proc_type>ens_fcst</gen_proc_type>
> @zhanglikate <https://github.com/zhanglikate> The "ens_fcst" is not an official type of generating process at
> https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-3.shtml <https://www.nco.ncep.noaa.gov/pmb/docs/grib2/grib2_doc/grib2_table4-3.shtml>
> Please revert it.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/NOAA-EMC/UPP/pull/768#pullrequestreview-1605495658>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APJPDRGSYZQVF6LGIIG6SKDXYDNHRANCNFSM6AAAAAA4GOXQPI>.
> You are receiving this because you were mentioned.
>

zhanglikate committed:
Update postcntrl_gefs_aerosol.xml postcntrl_gefs_aerosol_f00.xml

WenMeng-NOAA commented:
@zhanglikate You may add new aerosol fields in  the files postcntrl_gefs_aerosol*. I am meaning:
in postcntrl_gefs_aerosol.xml:
   Change
       <gen_proc_type>ens_fcst</gen_proc_type>
   into
        <gen_proc_type>fcst</gen_proc_type>

    change
         <gen_proc>gefs</gen_proc>
     into 
          <gen_proc>gfs_avn</gen_proc>

And the same changes for postcntrl_gefs_aerosol_f00.xml

WenMeng-NOAA commented on parm/postcntrl_gefs_aerosol.xml:
```diff
@@ -13,61 +13,994 @@
     <time_range_unit>hour</time_range_unit>
     <orig_center>nws_ncep</orig_center>
-    <gen_proc>gfs_avn</gen_proc>
+    <gen_proc>gfs</gen_proc>
```
@zhanglikate Please revert this change.

zhanglikate commented:
I just updated and committed. Can you check?


> On Aug 31, 2023, at 1:21 PM, WenMeng-NOAA ***@***.***> wrote:
> 
> 
> @zhanglikate <https://github.com/zhanglikate> You may add new aerosol fields in the files postcntrl_gefs_aerosol*. I am meaning:
> in postcntrl_gefs_aerosol.xml:
> Change
> <gen_proc_type>ens_fcst</gen_proc_type>
> into
> <gen_proc_type>fcst</gen_proc_type>
> 
> change
>      <gen_proc>gefs</gen_proc>
>  into 
>       <gen_proc>gfs_avn</gen_proc>
> And the same changes for postcntrl_gefs_aerosol_f00.xml
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/NOAA-EMC/UPP/pull/768#issuecomment-1701650571>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APJPDRHNEGUPLMBDUSRIVHDXYDP4ZANCNFSM6AAAAAA4GOXQPI>.
> You are receiving this because you were mentioned.
>

WenMeng-NOAA commented on parm/postcntrl_gefs_aerosol_f00.xml:
```diff
@@ -0,0 +1,1004 @@
+    <gen_proc_type>fcst</gen_proc_type>
+    <time_range_unit>hour</time_range_unit>
+    <orig_center>nws_ncep</orig_center>
+    <gen_proc>gfs</gen_proc>
```
@zhanglikate Please change "gfs" into "gfs_avn".

zhanglikate committed:
Update parm/postcntrl_gefs_aerosol.xml
parm/postcntrl_gefs_aerosol_f00.xml

zhanglikate commented:
Corrected. 

> On Aug 31, 2023, at 1:27 PM, WenMeng-NOAA ***@***.***> wrote:
> 
> 
> @WenMeng-NOAA commented on this pull request.
> 
> In parm/postcntrl_gefs_aerosol_f00.xml <https://github.com/NOAA-EMC/UPP/pull/768#discussion_r1312125415>:
> 
> > +<?xml version="1.0" encoding="UTF-8"?>
> +<postxml xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="EMC_POST_CTRL_Schema.xsd">
> +<!--  Lin Gan:  20160315 validated -->
> +  <paramset>
> +    <datset>GFSPRS</datset>
> +    <grid_num>0</grid_num>
> +    <sub_center>ncep_nco</sub_center>
> +    <version_no>v2003</version_no>
> +    <local_table_vers_no>local_tab_yes1</local_table_vers_no>
> +    <sigreftime>fcst</sigreftime>
> +    <prod_status>oper</prod_status>
> +    <data_type>fcst</data_type>
> +    <gen_proc_type>fcst</gen_proc_type>
> +    <time_range_unit>hour</time_range_unit>
> +    <orig_center>nws_ncep</orig_center>
> +    <gen_proc>gfs</gen_proc>
> @zhanglikate <https://github.com/zhanglikate> Please change "gfs" into "gfs_avn".
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/NOAA-EMC/UPP/pull/768#pullrequestreview-1605562806>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APJPDRBVV3FETXXSR63M2V3XYDQQ5ANCNFSM6AAAAAA4GOXQPI>.
> You are receiving this because you were mentioned.
>

WenMeng-NOAA commented:
@zhanglikate and @lipan-NOAA Do you have UFS model output  for the UPP standalone test? The files either on WCOSS2 or Hera would work for me.

zhanglikate commented:
> @zhanglikate and @lipan-NOAA Do you have UFS model output for the UPP standalone test? The files either on WCOSS2 or Hera would work for me.

@WenMeng-NOAA This the model results at HERA: /scratch1/BMC/gsd-fv3-dev/lzhang/AODtest/case2

WenMeng-NOAA commented:
@zhanglikate To generate GEFSV13 aerosol products in UPP, which option should I use in my test, gccpp_on, nasa_on, rdaod?

zhanglikate commented:
nasa_on. Thanks

> On Sep 1, 2023, at 11:07 AM, WenMeng-NOAA ***@***.***> wrote:
> 
> 
> @zhanglikate <https://github.com/zhanglikate> To generate GEFSV13 aerosol products in UPP, which option should I use in my test, gccpp_on, nasa_on, rdaod?
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/NOAA-EMC/UPP/pull/768#issuecomment-1703073410>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APJPDRFJL6J3URM7W64XXXDXYII6TANCNFSM6AAAAAA4GOXQPI>.
> You are receiving this because you were mentioned.
>

lipan-NOAA commented:
example file => HPSS: /NCEPDEV/emc-naqfc/1year/Li.Pan/tmp/forkate-atmnc-upptest-20180808.htar & forkate-sfcnc-upptest-20180808.htar

WenMeng-NOAA commented:
@zhanglikate You may add the following changes in parm/postcntrl_gefs_aerosol.xml and parm/postcntrl_gefs_aerosol_f00.xml to label ensemble products:
```
change
    <gen_proc_type>fcst</gen_proc_type>
into
   <gen_proc_type>ens_fcst</gen_proc_type>
change
    <gen_proc>gfs_avn</gen_proc>
into
   <gen_proc>gefs</gen_proc>
    
```
After update, the relevant txt files need to regenerated.

zhanglikate committed:
Move optics look up table from parm/ to fix/chem
Revert postcntrl_gefs_aerosol.xml and postcntrl_gefs_aerosol_f00.xml
Update makefile to include UFS-Aerosols for GEFSv13

zhanglikate commented:
> ens_fcst

@WenMeng-NOAA  Finished. Can you check? Thanks.

WenMeng-NOAA commented:
> > ens_fcst
> 
> @WenMeng-NOAA Finished. Can you check? Thanks.

@zhanglikate Your PR look good for me. I will run testing now. Thanks!

WenMeng-NOAA commented:
@zhanglikate and @lipan-NOAA Do you know if the file parm/AEROSOL_LUTS.dat is used for processing UFS based aerosol data? If not, can we delete it?

zhanglikate commented:
These are for GEFS-Aerosols, better to keep them. Thanks.

Kate

On Fri, Sep 8, 2023 at 8:06 AM WenMeng-NOAA ***@***.***>
wrote:

> @zhanglikate <https://github.com/zhanglikate> and @lipan-NOAA
> <https://github.com/lipan-NOAA> Do you know if the file
> parm/AEROSOL_LUTS.dat is used for processing UFS based aerosol data? If
> not, can we delete it?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/UPP/pull/768#issuecomment-1711730204>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APJPDRGPGGRW7AKPWP2XOG3XZMQ5DANCNFSM6AAAAAA4GOXQPI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

WenMeng-NOAA commented:
@zhanglikate My test is at /scratch1/NCEPDEV/stmp2/Wen.Meng/post_gefs_2018080800 on Hera. Please let me know if you see any discrepancies with your test.

lipan-NOAA commented:
I tested it on WCOSS2, it looks fine for me

WenMeng-NOAA commented:
> I tested it on WCOSS2, it looks fine for me

@lipan-NOAA Can you provide me the model output files in netcdf from your test on WCOSS2?

zhanglikate committed:
Delete parm/AEROSOL_LUTS.dat

WenMeng-NOAA commented:
@FernandoAndrade-NOAA You may start the UPP RT on Hera and Orion. There should be updates in job scripts of gefs testing with aerosol fix file relocation. You may pick up my changes from
```
Hera: /scratch2/NCEPDEV/ovp/Wen.Meng/ncep_post/post_regression_test_new/jobs
         run_post_fv3gefs_HERA.sh and run_post_fv3gefs_pe_test_HERA.sh

Orion: /home/wmeng/ovp/ncep_post/post_regression_test_new/jobs
         run_post_fv3gefs_ORION.sh and run_post_fv3gefs_pe_test_ORION.sh
```

WenMeng-NOAA commented:
The UPP RT was completed on WCOSS2 without changed results.

FernandoAndrade-NOAA commented:
> @FernandoAndrade-NOAA You may start the UPP RT on Hera and Orion. There should be updates in job scripts of gefs testing with aerosol fix file relocation. You may pick up my changes from
> 
> ```
> Hera: /scratch2/NCEPDEV/ovp/Wen.Meng/ncep_post/post_regression_test_new/jobs
>          run_post_fv3gefs_HERA.sh and run_post_fv3gefs_pe_test_HERA.sh
> 
> Orion: /home/wmeng/ovp/ncep_post/post_regression_test_new/jobs
>          run_post_fv3gefs_ORION.sh and run_post_fv3gefs_pe_test_ORION.sh
> ```

Apologies for the delay, I have RTs running on Hera and Orion. They've been temperamental today and dropping connection / freezing a lot from my side. I'll post results when they finish.

FernandoAndrade-NOAA commented:
UPP RTs have completed successfully on Hera and Orion with no changes to results.

WenMeng-NOAA commented:
This PR is ready for merging.
